bpf/lib/sock.h

```diff
@@ -16,3 +16,24 @@ __sock_cookie sock_local_cookie(struct bpf_sock_addr *ctx)
 #endif
 }
 
+#ifdef ENABLE_IPV4
+struct {
```

Comment:
The compiled elf files are used as the [template for loading and pinning maps](https://github.com/cilium/cilium/blob/main/pkg/bpf/collection.go#L286) by the agent.

Could moving this to another file have any implications there (i.e. by having it now show up in new elf files)?

cc @cilium/loader

Reply:
To my knowledge sock.h is only included into bpf_sock.c and now bpf_sock_term.c, so there shouldn't be any changes to other elf files.

Reply:
Somewhat related, I'm going to move the `struct sock_term_filter` and `struct cilium_sock_term_filter` definitions out of `sock.h` and into `bpf_sock_term.c` directly, so they aren't included in the `bpf_sock.c` elf.